### PR TITLE
Connect UI context cleanup

### PR DESCRIPTION
**Pull request by MananTank**
## PR-Codex overview
This PR focuses on cleaning up the Connect UI context code by replacing the usage of `useWalletConnectionCtx` with `useConnectUI`.

### Detailed summary
- Replaced `useWalletConnectionCtx` with `useConnectUI` in multiple files for context code cleanup.

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/web/wallets/shared/ScanScreen.tsx`, `packages/thirdweb/src/react/web/wallets/shared/ConnectingScreen.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/AnyWalletConnectUI.tsx`, `packages/thirdweb/src/react/core/providers/wallet-connection.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/screen.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx`, `packages/thirdweb/src/react/web/wallets/shared/WalletConnectConnection.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/AllWalletsUI.tsx`, `packages/thirdweb/src/react/web/wallets/in-app/InAppWalletSocialLogin.tsx`, `packages/thirdweb/src/react/web/wallets/in-app/InAppWalletSelectionUI.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/StartScreen.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/SmartWalletConnectUI.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Details.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectModalInline.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectModalContent.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectModal.tsx`, `packages/thirdweb/src/react/web/wallets/in-app/InAppWalletOTPLoginUI.tsx`, `packages/thirdweb/src/react/web/wallets/in-app/InAppWalletFormUI.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectEmbed.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/WalletSelector.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/ConnectWallet.tsx`, `packages/thirdweb/src/react/web/providers/wallet-ui-states-provider.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: .changeset/light-llamas-chew.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"thirdweb": patch
+---
+
+Connect UI context code cleanup
```
Comment (jnsdls) [inline .changeset/light-llamas-chew.md:5]: does this change any public api or fix a bug? if it's purely internal changes then we do not need a changeset for it

**Review by jnsdls — APPROVED** [review]
changes make sense - need to do manual testing since we don't have tests for this yet

File: .changeset/light-llamas-chew.md (hunk 1 above)
Comment (MananTank) [reply .changeset/light-llamas-chew.md:5]: No public API change - let me remove changeset

**Comment by MananTank** [comment]
> changes make sense - need to do manual testing since we don't have tests for this yet

Yep - did manual testing. You can try it out on https://internal-test-next-git-mnn-test-connect-ctx-refactor.thirdweb-preview.com/

**Pull request merged**